### PR TITLE
UX: Added deprecation warning for aurora

### PR DESCRIPTION
**Pull request by NidhiKJha**
When we remove Aurora as part of the reconfiguration of the popular custom network list, This PR is to create a warnning banner to inform our users about this change

I have converted the already existing DeprecatedTestNetworks component to the DeprecatedNetworks component so that we can use the same component every time we need to create a warning for new components.

## **Related issues**

Fixes: [#1605](https://github.com/MetaMask/MetaMask-planning/issues/1605)

## **Manual testing steps**

1. Add Aurora mainnet network
2. Switch to Aurora
3. Look at the banner at the bottom informing about the deprecation of Aurora
4. Close the banner it should go away
5. Reload the extension, the banner should show up
6. If close the banner, go to setting page Banner should show up again

### **Before**

NA

### **After**

Screenshot 2023-11-22 at 10 50 07 PM

Screenshot 2023-11-22 at 10 50 21 PM

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/multichain/network-list-item/network-list-item.js [hunk 1]
```diff
@@ -136,4 +147,5 @@ NetworkListItem.propTypes = {
    * Represents if the network item should be keyboard selected
    */
   focus: PropTypes.bool,
+  isDeprecatedNetwork: PropTypes.bool,
```
Comment (darkwing) [inline ui/components/multichain/network-list-item/network-list-item.js:150]: We should add doc comments here in the future.

File: ui/components/ui/deprecated-networks/deprecated-networks.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  const currentChainID = useSelector(getCurrentChainId);
+  const [isShowingWarning, setIsShowingWarning] = useState(false);
+  const completedOnboarding = useSelector(getCompletedOnboarding);
+  const t = useI18nContext();
+  useEffect(() => {
+    if (completedOnboarding && currentChainID === '0x4e454152') {
+      setIsShowingWarning(true);
+    } else {
+      setIsShowingWarning(false);
+    }
```
Comment (darkwing) [inline ui/components/ui/deprecated-networks/deprecated-networks.js:24]:
```
setIsShowingWarning(completedOnboarding && currentChainID === '0x4e454152')
```

File: ui/components/ui/deprecated-networks/deprecated-networks.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  }, [currentChainID, completedOnboarding]);
+
+  return (
+    isShowingWarning && (
```
Comment (darkwing) [inline ui/components/ui/deprecated-networks/deprecated-networks.js:28]: Can we use `condition ? <> : null` to be consistent?

File: ui/pages/routes/routes.component.js [hunk 4]
```diff
@@ -642,7 +642,7 @@ export default class Routes extends Component {
         })}
         dir={textDirection}
       >
-        {shouldShowNetworkDeprecationWarning && <DeprecatedTestNetworks />}
+        {shouldShowNetworkDeprecationWarning && <DeprecatedNetworks />}
```
Comment (darkwing) [inline ui/pages/routes/routes.component.js:645]: `condition ? <> : null`

**Review by darkwing — CHANGES_REQUESTED** [review]
A few tiny nits but mostly perfect.

File: ui/components/multichain/network-list-item/network-list-item.js (hunk 1 above)
Comment (NidhiKJha) [reply ui/components/multichain/network-list-item/network-list-item.js:153]: Updated here https://github.com/MetaMask/metamask-extension/pull/21933/commits/30f9a976ec84ad9c31351a3b8717845af2d14717

**Comment by NidhiKJha** [comment]
> A few tiny nits but mostly perfect.

Updated here https://github.com/MetaMask/metamask-extension/pull/21933/commits/30f9a976ec84ad9c31351a3b8717845af2d14717

File: app/_locales/en/messages.json [hunk 5]
```diff
@@ -489,6 +489,9 @@
   "attributions": {
     "message": "Attributions"
   },
+  "auroraDeprecationWarning": {
+    "message": "MetaMask will be deprecating support for Aurora network"
+  },
```
Comment (DDDDDanica) [inline app/_locales/en/messages.json:494]: would it be more useful to add a universal message for the upcoming network we want to deprecate?
```suggestion
  "auroraDeprecationWarning": {
    "message": "MetaMask will be deprecating support for $1 network"
  },
```

File: ui/pages/routes/routes.component.js [hunk 6]
```diff
@@ -642,7 +642,7 @@ export default class Routes extends Component {
         })}
         dir={textDirection}
       >
-        {shouldShowNetworkDeprecationWarning && <DeprecatedTestNetworks />}
+        {shouldShowNetworkDeprecationWarning ? <DeprecatedNetworks /> : null}
```
Comment (DDDDDanica) [inline ui/pages/routes/routes.component.js:645]: we can use original syntax?

File: shared/constants/network.ts [hunk 7]
```diff
@@ -172,6 +172,7 @@ export const CELO_DISPLAY_NAME = 'Celo Mainnet';
 export const GNOSIS_DISPLAY_NAME = 'Gnosis';
 export const ZK_SYNC_ERA_DISPLAY_NAME = 'zkSync Era Mainnet';
 export const BASE_DISPLAY_NAME = 'Base Mainnet';
+export const AURORA_ETH_DISPLAY_NAME = 'Aurora';
```
Comment (NidhiKJha) [inline shared/constants/network.ts:175]: We need this, as the other display name is Aurora Mainnet and as per the figma copy we need Aurora here

File: app/_locales/en/messages.json (hunk 5 above)
Comment (NidhiKJha) [reply app/_locales/en/messages.json]: Updated here https://github.com/MetaMask/metamask-extension/pull/21933/commits/14a11f2cdb30b03e7437a1bf3eb685e135d8ff58

File: ui/pages/routes/routes.component.js (hunk 6 above)
Comment (NidhiKJha) [reply ui/pages/routes/routes.component.js:645]: I think we can avoid `&&` and use ternary operators here re: https://github.com/MetaMask/metamask-extension/pull/21933#discussion_r1402468795

**Review by DDDDDanica — APPROVED** [review]

**Review by zone-live — APPROVED** [review]

File: ui/components/multichain/network-list-item/network-list-item.js [hunk 8]
```diff
@@ -93,6 +100,14 @@ export const NetworkListItem = ({
             name
           )}
         </Text>
+        {isDeprecatedNetwork ? (
+          <Tooltip
+            title={t('auroraDeprecationWarning', [AURORA_ETH_DISPLAY_NAME])}
```
Comment (darkwing) [inline ui/components/multichain/network-list-item/network-list-item.js:105]: I think this is fine for now but we'll eventually want a better system for allowing more than one deprecated network.

**Review by darkwing — APPROVED** [review]

File: ui/components/multichain/network-list-item/network-list-item.js (hunk 8 above)
Comment (NidhiKJha) [reply ui/components/multichain/network-list-item/network-list-item.js:105]: Noted! Yeah, so for the future we have plans to add a deprecated key in the networks list and update this file based on that. cc @kevinghim

**Pull request merged**